testing/cli/test_cli_sct_compute_snr.py

```diff
@@ -1,18 +1,144 @@
+    data1 = skimage.util.random_noise(data, mode='gaussian', clip=False, mean=0, var=1)
+    data2 = skimage.util.random_noise(data, mode='gaussian', clip=False, mean=0, var=1)
+    # Compute the square root of the sum of squares to obtain a Rayleigh (equivalent to Chi) distribution. This
+    # distribution is a more realistic representation of noise in magnitude MRI data, which is obtained by combining
+    # imaginary and real channels (each having Gaussian distribution).
+    data = np.sqrt(data1**2 + data2**2)
+    return data
+
+
+@pytest.fixture(scope="session")
+def dummy_3d_nib():
+    nii = nibabel.nifti1.Nifti1Image(dummy_3d_data(), np.eye(4))
+    filename = tempfile.NamedTemporaryFile(suffix='.nii.gz', delete=False).name
+    nibabel.save(nii, filename)
+    return filename
+
+
+@pytest.fixture(scope="session")
+def dummy_4d_nib():
+    # Create 4D volume. We need sufficient volumes to compute reliable standard deviation along the 4th dimension.
+    data = np.stack([dummy_3d_data() for i in range(50)], axis=3)
+    affine = np.eye(4)
+    nib = nibabel.nifti1.Nifti1Image(data, affine)
+    filename = tempfile.NamedTemporaryFile(suffix='.nii.gz', delete=False).name
+    nibabel.save(nib, filename)
+    return filename
+
+
+@pytest.fixture(scope="session")
+def dummy_3d_mask_nib():
+    data = np.zeros([32, 32, 32], dtype=np.uint8)
+    data[9:24, 9:24, 9:24] = 1
+    nii = nibabel.nifti1.Nifti1Image(data, np.eye(4))
+    filename = tempfile.NamedTemporaryFile(suffix='.nii.gz', delete=False).name
+    nibabel.save(nii, filename)
+    return filename
+
+
+@pytest.fixture(scope="session")
+def dummy_3d_mask_noise_nib():
+    data = np.zeros([32, 32, 32], dtype=np.uint8)
+    data[0:5, 0:5, 0:32] = 1
+    nii = nibabel.nifti1.Nifti1Image(data, np.eye(4))
+    filename = tempfile.NamedTemporaryFile(suffix='.nii.gz', delete=False).name
+    nibabel.save(nii, filename)
+    return filename
+
+
+def test_sct_compute_snr_check_dimension(dummy_3d_nib):
+    with pytest.raises(ValueError):
+        sct_compute_snr.main(argv=['-i', dummy_3d_nib, '-m', dummy_3d_nib, '-method', 'diff', '-vol', '0,5'])
+    with pytest.raises(ValueError):
+        sct_compute_snr.main(argv=['-i', dummy_3d_nib, '-m', dummy_3d_nib, '-method', 'mult', '-vol', '0,5'])
+
+
+def test_sct_compute_snr_check_dimension_mask(dummy_4d_nib):
+    with pytest.raises(ValueError):
+        sct_compute_snr.main(argv=['-i', dummy_4d_nib, '-m', dummy_4d_nib, '-method', 'mult'])
+
+
+def test_sct_compute_snr_check_vol_param(dummy_4d_nib, dummy_3d_nib):
+    with pytest.raises(ValueError):
+        sct_compute_snr.main(argv=['-i', dummy_4d_nib, '-m', dummy_3d_nib, '-m-noise', dummy_3d_nib,
+                                   '-vol', '0,1,2', '-method', 'single'])
+
+
+def test_sct_compute_snr_missing_mask(dummy_4d_nib, dummy_3d_nib):
+    with pytest.raises(RuntimeError):
+        sct_compute_snr.main(argv=['-i', dummy_4d_nib, '-m', dummy_3d_nib, '-method', 'single'])
```

Comment:
> but check the message as well. 

yeah... i'm afraid it would add unnecessary time on the core dev team in case someone slightly adjusts the message, then the test would not pass, the person would need to figure out why, etc.

Reply:
That's a good point... 

Maybe we wouldn't put the entire message, but a small part of the message that uniquely identifies that error. That way, minor changes can be made with no failure. For example, some of the messages mention a specific argument (e.g. `-m`) so `pytest.raises` could match against just that small string. 

I have a hunch that it might not be worth the effort, though. :thinking:

Reply:
sorry i have already spend too much time on this PR (other priorities are burning me)-- so i won't be able to follow up

Reply:
Of course. This is nitpicky anyway, so I will mark as resolved.